pkg/kn/root/root.go

```diff
@@ -86,6 +86,8 @@ Find more information about Knative at: https://knative.dev`, rootName),
 	rootCmd.PersistentFlags().StringVar(&p.KubeCfgPath, "kubeconfig", "", "kubectl configuration file (default: ~/.kube/config)")
 	rootCmd.PersistentFlags().StringVar(&p.KubeContext, "context", "", "name of the kubeconfig context to use")
 	rootCmd.PersistentFlags().StringVar(&p.KubeCluster, "cluster", "", "name of the kubeconfig cluster to use")
+	rootCmd.PersistentFlags().StringVar(&p.KubeAsUser, "as", "", "username to impersonate for the operation")
+	rootCmd.PersistentFlags().StringArrayVar(&p.KubeAsGroup, "as-group", []string{}, "group to impersonate for the operation, this flag can be repeated to specify multiple groups")
```

Comment:
I wonder if we should add `--as-uid` flag as well to complete the interface with `kubectl` options.

/cc @rhuss @vyasgun

Reply:
Oh interesting, I had never used that flag. That sounds like a good idea, will be adding shortly.

Reply:
@dsimansk looking at kubectl options, I only see as user and as groups for impersonation:

```
kubectl options | grep impersonate
      --as='': Username to impersonate for the operation
      --as-group=[]: Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
```

Reply:
Just added support for `as-uid`, which seems was added in 1.23